### ISSUE TITLE
Xamarin: Unable to use NativeMetadataCredentialsPlugin in AOT-only project

### ISSUE THREAD

**Issue by MaartenBrandemann** [grpc/grpc#18178]
### What version of gRPC and what language are you using?
 I am using the Grpc.Core (1.18.0), in combination with the Google.Cloud.Speech.V1(1.1.0) nuGet package. These I am using in Xamarin.Forms (C#).
 
### What operating system (Linux, Windows,...) and version?
 This crash happens when running the application on the iOS platform.
  
### What did you do?
When creating a Grpc.Core.Channel in the Xamarin.Forms shared project, I have the following exception. This is a iOS specific crash.

### What did you expect to see?
 I expected it to create a Grpc.Core.Channel, but this doesn't work on iOS. This seems to happen because the delegate in Grpc.Core.Internal.NativeMetadataCredentialsPlugin doesn't have the UnmanagedFunctionPointerAttribute (as explained in the link provided by the exception)
 
### What did you see instead?
```
Exception:
System.ExecutionEngineException: Attempting to JIT compile method '(wrapper native-to-managed) Grpc.Core.Internal.NativeMetadataCredentialsPlugin:NativeMetadataInterceptorHandler (intptr,intptr,intptr,intptr,intptr,int)' while running in aot-only mode. See https://docs.microsoft.com/xamarin/ios/internals/limitations for more information.
```

Stacktrace: 
```
Grpc.Core.Internal.NativeMetadataCredentialsPlugin..ctor(Grpc.Core.AsyncAuthInterceptor interceptor) in T:\src\github\grpc\src\csharp\Grpc.Core\Internal\NativeMetadataCredentialsPlugin.cs:47
Grpc.Core.MetadataCredentials.ToNativeCredentials() in T:\src\github\grpc\src\csharp\Grpc.Core\CallCredentials.cs:80
Grpc.Core.CompositeChannelCredentials.CreateNativeCredentials() in T:\src\github\grpc\src\csharp\Grpc.Core\ChannelCredentials.cs:202
Grpc.Core.ChannelCredentials..ctor() in T:\src\github\grpc\src\csharp\Grpc.Core\ChannelCredentials.cs:45
System.Lazy`1[T].CreateValue() in <e52fabd7a0824ae79d729c82e53f2aa9>:0
System.Lazy`1[T].LazyInitValue() in <e52fabd7a0824ae79d729c82e53f2aa9>:0
System.Lazy`1[T].get_Value() in <e52fabd7a0824ae79d729c82e53f2aa9>:0
 
Grpc.Core.ChannelCredentials.GetNativeCredentials() in T:\src\github\grpc\src\csharp\Grpc.Core\ChannelCredentials.cs:80
Grpc.Core.Channel..ctor(System.String target, Grpc.Core.ChannelCredentials credentials, System.Collections.Generic.IEnumerable`1[T] options) in T:\src\github\grpc\src\csharp\Grpc.Core\Channel.cs:77
Grpc.Core.Channel..ctor(System.String host, System.Int32 port, Grpc.Core.ChannelCredentials credentials, System.Collections.Generic.IEnumerable`1[T] options) in T:\src\github\grpc\src\csharp\Grpc.Core\Channel.cs:109
Grpc.Core.Channel..ctor(System.String host, System.Int32 port, Grpc.Core.ChannelCredentials credentials) in T:\src\github\grpc\src\csharp\Grpc.Core\Channel.cs:97
```
Thanks in advance,

Maarten

**Comment by jtattermusch** [comment]
The link https://docs.microsoft.com/xamarin/ios/internals/limitations

**Comment by jtattermusch** [comment]
This is a similar problem as https://github.com/grpc/grpc/issues/17587.

**Comment by jtattermusch** [comment]
I think this was fixed by  #18327. The fix will be in the upcoming release v1.20.0 (or you can test using nightly build packages from packages.grpc.io, they should be available soon).
Can you confirm that this was fixed?

**Comment by MaartenBrandemann** [comment]
@jtattermusch It seems to work! Thanks for helping out.
